### PR TITLE
CNFT1-2776 API: Create Extended patient data entry form for Ethnicity

### PR DESCRIPTION
**Pull request by stevegsa**
## Description

Work in progres...

## Tickets

* [CNFT1-2776](https://cdc-nbs.atlassian.net/browse/CNFT1-2776)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2776]: https://cdc-nbs.atlassian.net/browse/CNFT1-2776?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/create/PatientCreateController.java [hunk 1]
```diff
@@ -145,7 +150,14 @@ public PatientIdentifier create(@RequestBody NewPatient newPatient) {
         identificationService.add(context, newPatientIdentificationInput);
       });
     }
-
+    if (newPatient.ethnicity() != null) {
+      EthnicityInput ethnicityInput;
```
Comment (adamloup-enquizit) [inline apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/create/PatientCreateController.java:154]: question:  This is converting `EthnicityDemographic` -> `EthnicityInput` -> `PatientCommand.UpdateEthnicityInfo` would it make sense to include this in [PatientCreationService](https://github.com/CDCgov/NEDSS-Modernization/blob/main/apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/create/PatientCreationService.java) we can skip one transformation and go from `EthnicityDemographic` -> `PatientCommand.UpdateEthnicityInfo`

Comment (stevegsa) [reply apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/create/PatientCreateController.java:154]: Yes.  Let me pull the latest and integrate with that...

File: apps/modernization-api/src/test/java/gov/cdc/nbs/patient/profile/ethnicity/EthnicityEntrySteps.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  public void i_am_entering_patient_ethnicity_as_of(final Instant asOf) {
+    this.input.active(new EthnicityDemographic(asOf, null, null, null));
+  }
+
+  @Given("I enter the ethnicity unknown reason {string}")
+  public void i_enter_the_patient_ethnicity_unknown_reason(final String unknownReason) {
+    this.input.active(current -> current.withUnknownReason(unknownReason));
+  }
+
+  @Given("I enter the ethnicity ethnic group {string}")
```
Comment (adamloup-enquizit) [inline apps/modernization-api/src/test/java/gov/cdc/nbs/patient/profile/ethnicity/EthnicityEntrySteps.java:26]: suggestion: Using `I enter the ethnicity ethnic group {ethnicity}` will let you write the step using values from the ethnic group value set like `Hispanic or Latino`, `Not Hispanic or Latino`, or `Unknown`.  The same applies to ethnicity_detailed, the cucumber parameter type for that value set is 'ethnicityDetail`

**Review by adamloup-enquizit — APPROVED** [review]
👍 Looks good!


💀 the formatting changes 💀

**Review by mpeels — APPROVED** [review]

**Pull request merged**